### ISSUE TITLE
Having to press Enter twice for paste to work

### ISSUE THREAD

**Issue by SirDufus** [sabrogden/Ditto#769]
I'm working away, press shortcut key to open Ditto window, use the down arrow key to find item I need from list, press enter. Almost all the time, it activates the search bard instead, but if I press Enter key again, will paste the information.

Most of the time, have to press twice, some of the time I only have to press once. I've tried to figure out a pattern or to tweak some setting but nothing I do helps.

Any suggestions on what I need to change in settings or am is that just how it works perhaps? Thanks!

**Comment by SirDufus** [comment]
Any ideas on this? Maybe I didn't explain myself correctly as English is my second language?

**Comment by sabrogden** [comment]
Try this

- Enable outputting debug to OutputDebugString, options - general - advanced - "Write debug to OutputDebugString" = True
- Download DebugView https://learn.microsoft.com/en-us/sysinternals/downloads/debugview
- Open Ditto and hit enter, see what is output in DebugView
- Post the contents back and we'll see what is in there

**Comment by SirDufus** [comment]
Thanks for following up, appreciate any help you can offer so I only have to press enter once to paste. 

Here is the log as requested, also, changed name of site for privacy reasons but everything else is exactly the same:

[11176] [2025/1/24 06:33:43.108 - MainFrm.cpp 303] On Show Ditto HotKey, showing window
[11176] [2025/1/24 06:33:43.108 - ExternalWindowTracker.cpp 100] TargetActiveWindow Active: SiteNameRemoved - Google Chrome (328540), Focus: SiteNameRemoved - Google Chrome (328540), FromHook 1, IdleTime: 0.000000
[11176] [2025/1/24 06:33:43.108 - QuickPaste.cpp 80] Start of ShowQPasteWnd, AtPrevPos: 0, FromKeyboard: 1, RefillList: 0
[11176] [2025/1/24 06:33:43.112 - QPasteWnd.cpp 746] CQPasteWnd::OnActivate, nState: 1, Other: 0, Minimized: 0
[11176] [2025/1/24 06:33:43.112 - QPasteWnd.cpp 922] Start - ShowQPasteWindow - Fill List: 1, array count: 0
[11176] [2025/1/24 06:33:43.112 - Theme.cpp 79] Loading theme based on windows setting of dark mode for apps
[11176] [2025/1/24 06:33:43.116 - QPasteWnd.cpp 1374] Start Fill List - 
[11176] [2025/1/24 06:33:43.116 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 3 - Name CQPasteWndThread
[11176] [2025/1/24 06:33:43.116 - QPasteWndThread.cpp 36] Start of OnEvent, eventId: Load clips
[11176] [2025/1/24 06:33:43.116 - QPasteWndThread.cpp 123] Load Items start = 0, count = 16, list size: 0
[11176] [2025/1/24 06:33:43.116 - QPasteWnd.cpp 1581] Start Fill List - Count SQL: SELECT COUNT( Main.lID) FROM Main  where ((Main.bIsGroup = 1 AND Main.lParentID = -1) OR (Main.bIsGroup = 0 AND Main.lParentID = -1)), Query SQL: SELECT  Main.lID, Main.mText, Main.lParentID, Main.lDontAutoDelete, Main.lShortCut, Main.bIsGroup, Main.QuickPasteText, Main.clipOrder, Main.clipGroupOrder, Main.stickyClipOrder, Main.stickyClipGroupOrder, Main.lDate, Main.lastPasteDate FROM Main  where ((Main.bIsGroup = 1 AND Main.lParentID = -1) OR (Main.bIsGroup = 0 AND Main.lParentID = -1)) order by Main.stickyClipOrder DESC, Main.bIsGroup ASC, Main.clipOrder DESC
[11176] [2025/1/24 06:33:43.116 - QPasteWnd.cpp 958] END - ShowQPasteWindow - Fill List: 1, array count: 0
[11176] [2025/1/24 06:33:43.117 - QPasteWndThread.cpp 85] Set list count = 500, time = 0
[11176] [2025/1/24 06:33:43.117 - QPasteWnd.cpp 2882] SetKeyModiferState 1
[11176] [2025/1/24 06:33:43.117 - QuickPaste.cpp 288] END of ShowQPasteWnd, AtPrevPos: 0, FromKeyboard: 1, RefillList: 0, Position, 0 0 1843 846
[11176] [2025/1/24 06:33:43.117 - QPasteWndThread.cpp 229] Load items End count = 16, Total Time = 16, LoadItems: 16, Count: 0, Accel: 0
[11176] [2025/1/24 06:33:43.117 - QPasteWndThread.cpp 58] End of OnEvent, eventId: Load clips, Time: 16(ms)
[11176] [2025/1/24 06:33:43.117 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 3 - Name: 1388130072
[11176] [2025/1/24 06:33:43.228 - MainFrm.cpp 743] Timer KEY_STATE_MODIFIERS count NOT hit(20), count (2) time (125)
[11176] [2025/1/24 06:33:43.228 - QPasteWnd.cpp 2882] SetKeyModiferState 0
[11176] [2025/1/24 06:33:44.790 - ExternalWindowTracker.cpp 25] Not Idle for long enough, IdleTime: 0.031000, MinIdle 0.100000
[11176] [2025/1/24 06:33:46.464 - QPasteWnd.cpp 6171] TIMER_DO_SEARCH timer
[11176] 
[11176] [2025/1/24 06:33:46.466 - QPasteWnd.cpp 1374] Start Fill List -  
[11176] [2025/1/24 06:33:46.468 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 3 - Name CQPasteWndThread
[11176] [2025/1/24 06:33:46.468 - QPasteWndThread.cpp 36] Start of OnEvent, eventId: Load clips
[11176] [2025/1/24 06:33:46.468 - QPasteWndThread.cpp 123] Load Items start = 0, count = 16, list size: 0
[11176] [2025/1/24 06:33:46.468 - QPasteWnd.cpp 1581] Start Fill List - Count SQL: SELECT COUNT( Main.lID) FROM Main INNER JOIN Data on Data.lParentID = Main.lID where ((Data.strClipBoardFormat = 'CF_UNICODETEXT' AND Data.ooData LIKE '%%%%' ESCAPE '\')), Query SQL: SELECT  Main.lID, Main.mText, Main.lParentID, Main.lDontAutoDelete, Main.lShortCut, Main.bIsGroup, Main.QuickPasteText, Main.clipOrder, Main.clipGroupOrder, Main.stickyClipOrder, Main.stickyClipGroupOrder, Main.lDate, Main.lastPasteDate FROM Main INNER JOIN Data on Data.lParentID = Main.lID where ((Data.strClipBoardFormat = 'CF_UNICODETEXT' AND Data.ooData LIKE '%%%%' ESCAPE '\')) order by Main.stickyClipOrder DESC, Main.bIsGroup ASC, Main.clipOrder DESC
[11176] [2025/1/24 06:33:46.471 - QPasteWndThread.cpp 85] Set list count = 500, time = 0
[11176] [2025/1/24 06:33:46.471 - QPasteWndThread.cpp 229] Load items End count = 0, Total Time = 0, LoadItems: 0, Count: 0, Accel: 0
[11176] [2025/1/24 06:33:46.471 - QPasteWndThread.cpp 58] End of OnEvent, eventId: Load clips, Time: 0(ms)
[11176] [2025/1/24 06:33:46.471 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 3 - Name: 1388130072
[11176] [2025/1/24 06:33:46.791 - ExternalWindowTracker.cpp 78] Ditto has focus - Active: SiteNameRemoved - Google Chrome (328540), Focus: SiteNameRemoved - Google Chrome (328540), FromHook 1
[11176] [2025/1/24 06:33:54.793 - ExternalWindowTracker.cpp 25] Not Idle for long enough, IdleTime: 0.031000, MinIdle 0.100000
[11176] [2025/1/24 06:33:56.846 - QPasteWnd.cpp 746] CQPasteWnd::OnActivate, nState: 0, Other: 0, Minimized: 0
[11176] [2025/1/24 06:33:56.846 - QPasteWnd.cpp 817] Start of HideQPasteWindow
[11176] [2025/1/24 06:33:56.949 - QPasteWnd.cpp 901] End of HideQPasteWindow, ItemCount: 0
[11176] [2025/1/24 06:34:30.732 - EventThread.cpp 210] Start of CEventThread::RunThread() - OnEvent 4 - Name CMainFrmThread
[11176] [2025/1/24 06:34:30.732 - EventThread.cpp 212] End of CEventThread::RunThread() - OnEvent 4 - Name: 1388254616

**Comment by SirDufus** [comment]
Can anyone offer advice please?

**Comment by Chealer** [comment]
Thank you for reporting

Does this persist with 03.25.113 or ulterior?

If so, does this happen using Return? And does it happen if you double-click an entry with the mouse?

**Comment by SirDufus** [comment]
Thanks for following up! 

Yes the issue is still the same as reported. Am using a new computer with Windows 11 and Ditto v 03.25.113. While waiting for a fix,  I've had to use the shortcut workaround. To be clear, the issue is and has always been with pressing ENTER key, nothing has changed, behavior is same as previously described. 

Also, never use mouse to select entries but I did a few quick tests and no the issue is not present when using mouse as far as I can tell but again, only tried a few times.

**Comment by Chealer** [comment]
Thank you @SirDufus 

Are you saying that this affects 2 different computers you use?

Which keyboard layout do you use?

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by SirDufus** [comment]
Yes that is correct, this bug is existent on every device I install the app on. Currently have the same issue on Windows 10 or Windows 11. 

Keyboard is normal US layout..

**Comment by Chealer** [comment]
Thanks again @SirDufus 

So, both of the affected PCs have Windows set to use the QWERTY "English (USA)" keyboard layout? Can you see any hardware particularity shared by these PCs (for example, an identical keyboard model)?

> While waiting for a fix, I've had to use the shortcut workaround.

Please clarify what you mean by "the shortcut workaround"

**Comment by SirDufus** [comment]
Yes, same logitech keyboard on both, I'll try using just the laptop keyboard to see if that changes anything. 

Also, I've been playing around a bit and noticed that when I press enter and it doesn't work, what happens is that is selects and activates the search box at bottom of window, then when I press enter again, this time it pastest the text. 

Having said that, often, it just works as expected, and the search box doesn't get activated.

**Comment by Chealer** [comment]
Hum, interesting. As discussed in ticket #431, when Quick Paste opens, a clip should be preselected (generally the top one). Is that what happens when this issue occurs?

> Yes, same logitech keyboard on both, I'll try using just the laptop keyboard to see if that changes anything.

Please keep us updated, and indicate which model that is if it's specific to that keyboard.

**Comment by SirDufus** [comment]
Using Logitech MX Mechanical keyboard with Logi Options+ but the app has been removed from the system now which I thought had actually fixed the problem. But after hours of trying, first thing this morning it was back. 

I spent a few more hours today, again, thought it was fixed and just when I am about to walk away, it's back! Since it doesn't happen often, I suspect it was never solved to begin with and I was running on false hope lol. I've spent hours and hours trying to figure this out. Because I am unable to reproduce (it's always random and not often), it's starting to turn my hair grey. 

I've removed mouse and keyboard bloatware, so I know for sure it's not mouse or keyboard software causing the issue.

I'm out of ideas, all I got left is try to keep an eye out to see if it always pre-selects a clip as you requested.

**Comment by Chealer** [comment]
Thank you very much @SirDufus 
Do you have any other utility shared between these installs, like Microsoft PowerToys? If so, does this persist when these do not run?

**Comment by SirDufus** [comment]
Yes, they both use Power Toys. This morning, I power on Windows, disabled the Keyboard Manager plugin and the issue pop up almost right away after doing that. I'm going to try again tomorrow with Power Toys not running at all instead of disabling the plug in. I'm also going to install on another W11 machine that does not have Power toys and uses a different keyboard to see what happens over there.

To circle back to my earlier comment, where I mentioned using shortcuts, in the ditto options, under quick paste keyboard, there are options for paste position 1, paste position 2 and so on. I have those set to use Shift 0 to Shift 9 and they work on the first try every single time. That's what I've been using as a work around but it's not a perfect solution as I often have to go way past 10 entries to find what I'm looking for.

Thanks for giving me lots of places to check. What's odd is that it does always auto select the top entry when I open Ditto, also, when I use arrow down to choose an entry, that also always works on first try.

**Comment by Chealer** [comment]
Thank you very much @SirDufus 
Keyboard Manager would definitely be a prime suspect, but I use PowerToys on all my Ditto installs, and Keyboard Manager is enabled, although I never use the <kbd>Enter</kbd> key with Ditto.

> To circle back to my earlier comment, where I mentioned using shortcuts, in the ditto options, under quick paste keyboard, there are options for paste position 1, paste position 2 and so on. I have those set to use Shift 0 to Shift 9 and they work on the first try every single time. That's what I've been using as a work around but it's not a perfect solution as I often have to go way past 10 entries to find what I'm looking for.

Ah, that explains what you meant by "the shortcut workaround", thanks
